ompi/mca/common/ompio/common_ompio_buffer.c

```diff
@@ -82,8 +83,11 @@ static void* mca_common_ompio_buffer_alloc_seg ( void*ctx, size_t *size )
 
 static void mca_common_ompio_buffer_free_seg ( void *ctx, void *buf )
 {
+    int tmp = -1;
+    uint64_t flags = -1;
+
     if ( NULL != buf ) {
-        if (NULL != opal_accelerator.host_unregister) {
+        if (opal_accelerator.check_addr(buf, &tmp, &flags) > 0) {
```

Comment:
I think this check is wrong. buf is supposed to be a host buffer, not a device buffer. Read here:

https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1ge8d5c17670f16ac4fc8fcb4181cb490c

Reply:
removed commit